lib/discordrb/data/role.rb

```diff
@@ -183,6 +209,28 @@ def icon_url(format = 'webp')
       Discordrb::API.role_icon_url(@id, @icon, format)
     end
 
+    # Get the icon that a role has displayed.
+    # @return [String, nil] Icon URL, the unicode emoji, or nil if this role doesn't have any icon.
+    # @note A role can have a unicode emoji, and an icon, but only the icon will be shown in the UI.
+    def display_icon
+      icon_url || unicode_emoji
+    end
+
+    # Set the icon this role is displaying.
+    # @param icon [File, String, nil] File like object that responds to #read, unicode emoji, or nil.
+    # @note Setting the icon to nil will remove the unicode emoji **and** the custom icon.
+    def display_icon=(icon)
+      # rubocop:disable Lint/ReturnInVoidContext
+      return update_role_data(unicode_emoji: nil, icon: nil) if icon.nil?
+      # rubocop:enable Lint/ReturnInVoidContext
```

Comment:
I don't want to disable per line here, this could just as easily be reworked to a block that acts and then returns for more clarity that the return value isn't usable data

Also, if `icon` is nil, the below else block should run.
```
irb(main):001> nil.read
(irb):1:in `<main>': undefined method `read' for nil (NoMethodError)
```

Was this copied from somewhere else in the codebase?